src/main/op/OP.ts

```diff
@@ -59,8 +66,37 @@ export class OP {
     requestJwtOrUri: string | URI,
     requestOpts?: { nonce?: string; verification?: InternalVerification | ExternalVerification }
   ): Promise<VerifiedAuthorizationRequest> {
-    const authorizationRequest = await AuthorizationRequest.fromUriOrJwt(requestJwtOrUri);
-    return await authorizationRequest.verify(this.newVerifyAuthorizationRequestOpts({ ...requestOpts }));
+    const authorizationRequest = await AuthorizationRequest.fromUriOrJwt(requestJwtOrUri)
+      .then((result: AuthorizationRequest) => {
+        if (this._eventEmitter) {
```

Comment:
suggest to make a private method, that does the if check and emits if present. Then you get some nicer code instead of the 4 times `if (this._eventEmmiter)`

It could even retrhow the error if present. Meaning your code will always simply call emitEvent

So the body is doing something like this:

````typescript
ìf (this._eventEmmitter) {
   this._eventEmitter.emit(event)
}
if (event.hasError()) {
  throw event.getError()
}
````